py3status/modules/external_script.py

```diff
@@ -10,9 +10,11 @@
 The script should not have any parameters, but it could work.
 
 Configuration parameters:
+    button_refresh: button to refresh the module (default 2)
     cache_timeout: how often we refresh this module in seconds
         (default 15)
     format: see placeholders below (default '{output}')
+    format_notification: see placeholders below (default '{output}')
```

Comment:
This config need to be `None` by default so we don't annoy other users.

Reply:
Why would they be annoyed? Middle click (default button for refresh) still works as it was working before, the rest of the buttons had no action assigned to them, now they do something useful - show full output of the script.

Reply:
Everybody have different goals with their `external_script` and we don't know what their goals are. It can be something trivial like to clicking immediately to update a number of mails or updates... and not wanting to see a notification about last number of mails or updates... or to simply execute something periodically with static text... only to be notified about the same good ol' static text.

I assume 99% of time, it will be be same as the one they have on their bar. You want _New!_ full output in your notification. That's fine and we can add it... but not with default `notification = '{output}'` or `notification = '{output_full}'`.

 With `None` by default, we can quietly introduce this new feature without _literally_ notifying people about it.

Reply:
Clicking to refresh is what `button_refresh` is for, it doesn't show notifications and is there exactly for these cases. You said so yourself, it is good to keep `button_refresh` to `2` by default so that everyone knows: middle click is refresh with no extra actions. Other buttons do extra actions.

I don't see this as a feature "only for me", which I should hide from other users, I genuinely think it makes sense to be on by default. Same in battery_level by the way, I don't see why this is an opt-in feature, in my mind button `1` should always show notification and button `2` should be for silent refreshes.